api/profile/profileRouter.js

```diff
@@ -9,6 +9,15 @@ const {
 const { validateUser } = require('../middleware/generalMiddleware');
 validateUser;
 
+router.get('/current_user_profile/', authRequired, async (req, res) => {
+  try {
+    const resp = await Profiles.mentorApplicationData(req.profile.profile_id);
+    res.status(200).json(resp);
+  } catch (error) {
+    res.status(500).json({ message: error.message });
```

Comment:
I think you should call `next(err)` here to reference the app's primary error handler, instead of always sending a 500 status.

Reply:
good catch. I initially had some errors when trying to use next, so i looked at the error handling format of the other profile-router routes, and switched it out. I have added it back (but it appears that every backend commit has also appeared on here as well so i may just end up deleting this branch and submitting a new PR)